control-plane/agents/common/src/msg_translation/v1.rs

```diff
@@ -334,3 +334,42 @@ pub fn rpc_nexus_to_child_agent(
         id: child_uri,
     })
 }
+
+impl IoEngineToAgent for v1_rpc::pool::Pool {
+    type AgentMessage = transport::PoolState;
+    ///This converts gRPC pool object into Control plane Pool state.
+    fn to_agent(&self) -> Self::AgentMessage {
+        Self::AgentMessage {
+            node: Default::default(),
+            id: self.name.clone().into(),
+            disks: self.disks.clone().into_vec(),
+            capacity: self.capacity,
+            used: self.used,
+            status: self.state.into(),
+        }
+    }
+}
+
+impl AgentToIoEngine for transport::CreatePool {
+    type IoEngineMessage = v1_rpc::pool::CreatePoolRequest;
+    ///This converts Control plane CreatePool struct to IO Engine gRPC message.
+    fn to_rpc(&self) -> Self::IoEngineMessage {
+        Self::IoEngineMessage {
+            name: self.id.clone().into(),
+            disks: self.disks.iter().map(|d| d.to_string()).collect(),
+            uuid: None,
+            pooltype: 0,
+        }
+    }
+}
+
+impl AgentToIoEngine for transport::DestroyPool {
+    type IoEngineMessage = v1_rpc::pool::DestroyPoolRequest;
+    ///This converts Control plane DeletePool struct to IO Engine gRPC message.
```

Comment:
here and above, please add space between /// and first letter.

Reply:
Incorporated